sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileAsyncClient.java

```diff
@@ -307,7 +307,7 @@ public String getFileName() {
         return chunkedSource.concatMap(pool::write)
             .concatWith(Flux.defer(pool::flush))
             /* Map the data to a tuple 3, of buffer, buffer length, buffer offset */
-            .map(buffer -> Tuples.of(buffer, buffer.remaining(), 0L))
+            .map(bufferAggregator -> Tuples.of(bufferAggregator, bufferAggregator.length(), 0L))
```

Comment:
Should have used .lenght() everywhere in this  method after transition from ByteBuffer to BufferAggregator.
Confusing method has been hidden and renamed...

Reply:
Sweet great to hear we found something!